docs/src/main/sphinx/admin/properties-optimizer.rst

```diff
@@ -117,6 +117,16 @@ the maximum number of joins that can be reordered at once.
 Reduces number of rows produced by joins when optimizer detects that duplicated
 join output rows can be skipped.
 
+``optimizer.use-exact-partitioning``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-boolean`
+* **Default value:** ``false``
+
+Re-partitions data unless the partitioning of the upstream stage exactly matches
+what the downstream stage expects. This can also be specified on a per-query
```

Comment:
Re-partition data unless 

also .. do we want to link to concepts page for stages or is this assumed knowledge on this page .. I kinda think yes